frontend/src/calculations/FieldAndSoil/Crops/Calculations.ts

```diff
@@ -407,3 +407,194 @@ export async function getCropRequirementP205(
+      }
+    } else if (leafTissueP > 0.19) {
+      if (soilTestValP < 15) {
+        tempReqP2O5 = 40;
+      } else if (soilTestValP >= 15 && soilTestValP <= 30) {
+        tempReqP2O5 = 30;
+      } else if (soilTestValP > 30) {
+        tempReqP2O5 = 0;
+      }
+    }
+    nutrientInputs.reqP2o5 = Math.round(tempReqP2O5);
+
+    // Calculate K2O requirement based on leaf tissue K and soil test K
+    let tempReqK2O = 0;
+    if (leafTissueK < 1.0) {
+      if (soilTestValK < 120) {
+        tempReqK2O = 100;
+      } else if (soilTestValK >= 120 && soilTestValK <= 280) {
+        tempReqK2O = 80;
+      } else if (soilTestValK > 280) {
+        tempReqK2O = 50;
+      }
+    } else if (leafTissueK >= 1.0 && leafTissueK <= 1.25) {
+      if (soilTestValK < 120) {
+        tempReqK2O = 80;
+      } else if (soilTestValK >= 120 && soilTestValK <= 280) {
+        tempReqK2O = 60;
+      } else if (soilTestValK > 280) {
+        tempReqK2O = 30;
+      }
+    } else if (leafTissueK > 1.25) {
+      if (soilTestValK < 120) {
+        tempReqK2O = 50;
+      } else if (soilTestValK >= 120 && soilTestValK <= 280) {
+        tempReqK2O = 30;
+      } else if (soilTestValK > 280) {
+        tempReqK2O = 0;
+      }
+    }
+    nutrientInputs.reqK2o = Math.round(tempReqK2O);
+
+    // Calculate P2O5 removal
+    let tempRemP2O5 = cropYield;
+    const isPrunedAndRemoved = willPlantsBePruned && whereWillPruningsGo === 'Removed from field';
+    tempRemP2O5 = tempRemP2O5 * 1.145 + (isPrunedAndRemoved ? 2.748 : 0);
+    nutrientInputs.remP2o5 = Math.round(tempRemP2O5);
+
+    // Calculate K2O removal
+    let tempRemK2O = cropYield;
+    tempRemK2O = tempRemK2O * 3.63 + (isPrunedAndRemoved ? 11.374 : 0);
+    nutrientInputs.remK2o = Math.round(tempRemK2O);
+
+    return nutrientInputs;
+  } catch (error) {
+    console.error('Error calculating crop requirement removal for raspberries:', error);
+    return nutrientInputs;
+  }
+}
+
+export async function getBlueberryNutrients(
+  cropYield: number,
+  willSawdustBeApplied: boolean,
+  willPlantsBePruned: boolean,
+  whereWillPruningsGo: string,
+  plantAgeYears: string,
+  numberOfPlantsPerAcre: number,
+  soilTestValP: number,
+  leafTissueP: number,
+  leafTissueK: number,
+) {
+  const nutrientInputs = {
+    reqN: 0,
+    reqP2o5: 0,
+    reqK2o: 0,
+    remN: 0,
+    remP2o5: 0,
+    remK2o: 0,
+  };
+
+  try {
+    // Nitrogen calculation based on plant age
+    const NGmPlantArr = [
+      { key: '1', value: 6 },
+      { key: '2', value: 8.5 },
+      { key: '3', value: 14 },
+      { key: '4', value: 23 },
+      { key: '5', value: 28 },
+      { key: '6', value: 31 },
+      { key: '7', value: 40 },
+      { key: '8', value: 45 },
+      { key: '9 or more', value: 63 },
+    ];
+
+    const plantAge = plantAgeYears;
+    const plantsPerAcre = numberOfPlantsPerAcre || 0;
+    const tempN = NGmPlantArr.find((item) => item.key === plantAge)?.value || 0;
+
+    nutrientInputs.reqN = Math.round(
```

Comment:
Plz add equation comments as part of your cleanup cause I don't get what's happening, unless the original equations truly don't have a clear rhyme or reason